libraries/api-extractor/src/PackageJsonLookup.ts

```diff
@@ -0,0 +1,76 @@
+/* tslint:disable:no-constant-condition */
+
+import * as fsx from 'fs-extra';
+import * as path from 'path';
+import JsonFile from './JsonFile';
+
+/**
+ * Represents a package.json file.
+ */
+interface IPackageJson {
+  name: string;
+}
+
+/**
+ * This class provides methods for finding the nearest "package.json" for a folder
+ * and retrieving the name of the package.  The results are cached.
+ */
+export default class PackageJsonLookup {
```

Comment:
This code is already heavily covered by the other unit tests, but I can add a dedicated test as well.

---
In reply to: [122495182](https://github.com/Microsoft/web-build-tools/pull/236#discussion_r122495182) [](ancestors = 122495182)